src/aura/TasksTableComponent/TasksTableComponentHelper.js

```diff
@@ -77,4 +76,27 @@
 
 	},
 
-})
+	refreshTasksList : function(component, task){
+
+
+		if(task.RecordType.Name==='TO-DO')
+		{
+			var TaskTODOlist = component.get('v.ToDoTasks');
+			var newList = TaskTODOlist.filter((a) => a.Status__c != 'Completed');
```

Comment:
Your function should actually filter the items such as "a.Id != task.Id", because you already know which element you want to remove. Please adjust this and other examples in this same method.

Reply:
Ready